Framework/Algorithms/src/CalculateFlatBackground.cpp

```diff
@@ -117,9 +118,7 @@ void CalculateFlatBackground::exec() {
     PARALLEL_FOR2(inputWS, outputWS)
     for (int i = 0; i < numHists; ++i) {
       PARALLEL_START_INTERUPT_REGION
-      outputWS->dataX(i) = inputWS->readX(i);
-      outputWS->dataY(i) = inputWS->readY(i);
-      outputWS->dataE(i) = inputWS->readE(i);
+      outputWS->setHistogram(i, inputWS->histogram(i));
```

Comment:
We have to be careful here. If `inputWS` is an `EventWorkspace`, does `histogram()` return the full histogram, or does it have `NULL` `Y` and `E`?

The question is not actually about this particular example, but in general;

- Should `MatrixWorkspace::histogram()` return a fully initialized `Histogram`? I think the answer should be yes.
- Given the above, do we provide a method for extracting the partial `Histogram` from an `EventWorkspace`?